library/alloc/src/sync.rs

```diff
@@ -1346,18 +1346,18 @@ impl<T: ?Sized> Receiver for Arc<T> {}
 impl<T: Clone> Arc<T> {
     /// Makes a mutable reference into the given `Arc`.
     ///
-    /// If there are other `Arc` or [`Weak`] pointers to the same allocation,
-    /// then `make_mut` will create a new allocation and invoke [`clone`][clone] on the inner value
-    /// to ensure unique ownership. This is also referred to as clone-on-write.
+    /// If there are other `Arc` pointers to the same allocation, then `make_mut` will
+    /// [`clone`] the inner value to a new allocation to ensure unique ownership.  This is also
+    /// referred to as clone-on-write.
     ///
-    /// Note that this differs from the behavior of [`Rc::make_mut`] which disassociates
-    /// any remaining `Weak` pointers.
+    /// However, if there are no other `Arc` pointers to this allocation, but some [`Weak`]
+    /// pointers, then the [`Weak`] pointers will be disassociated and the inner value will not
+    /// be cloned.
     ///
-    /// See also [`get_mut`][get_mut], which will fail rather than cloning.
+    /// See also [`get_mut`], which will fail rather than cloning.
```

Comment:
Hm - I think this may now be a little misleading. It looks like get_mut will not "drop" Weaks, so it might be worth clarifying here (e.g., "which will fail with any existing references, both weak and strong")

Reply:
Maybe “which will fail rather than cloning the inner value or diassociating [`Weak`] pointers”?

Reply:
`get_mut` has a similarly inaccurate sentence
> See also [`make_mut`], which will [`clone`] the inner value when there are other pointers.

Reply:
Perhaps change that to ``See also [`make_mut`], which will [`clone`] the inner value when there are other [`Arc`] pointers.`` (or ``[`Rc`]``, respectively)

Reply:
Your suggested phrasing seems good.

I think maybe a good alternative is to make sure each of get_mut/make_mut is clear about the conditions, and then just say "See also $other_one" and not try to describe what it does in a sentence.

Reply:
Made a new commit.

Reply:
I don’t think it hurts necessarily to have the quick comparing explanation; especially since it’s pointing out a relevant difference between the two.

Reply:
And each one by itself should be clear about the conditions right now.